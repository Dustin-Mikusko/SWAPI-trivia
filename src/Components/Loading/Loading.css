@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400&display=swap');

.loader {
  height: 100vh;
  width: 100%;
  background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.7)), url('../../images/background-stars.jpeg');
  display: flex;
  justify-content: center;
  align-items: center;
}



.loading-title {
  color: #08d5d5;
  font-size: 4em;
  font-family: 'Open Sans', sans-serif;
  font-weight: 400;
}



/* loading dots */

.loading:after {
  content: ' .';
  animation: dots 2s steps(5, end) infinite;}

@keyframes dots {
  0%, 20% {
    color: rgba(0,0,0,0);
    text-shadow:
      .25em 0 0 rgba(0,0,0,0),
      .5em 0 0 rgba(0,0,0,0);}
  40% {
    color: #08d5d5;
    text-shadow:
      .25em 0 0 rgba(0,0,0,0),
      .5em 0 0 rgba(0,0,0,0);}
  60% {
    text-shadow:
      .25em 0 0 #08d5d5,
      .5em 0 0 rgba(0,0,0,0);}
  80%, 100% {
    text-shadow:
      .25em 0 0 #08d5d5,
      .5em 0 0 #08d5d5;}}
