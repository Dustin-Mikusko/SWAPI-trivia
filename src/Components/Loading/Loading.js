import React from 'react'
import './Loading.css';

const Loading = () => {
  return (
    <main className="loader">
      <div className="title-container">
        <p className="loading-title loading">Gathering data from a galaxy far, <br />far away</p>
      </div>
    </main>
    
  )
}

export default Loading;
