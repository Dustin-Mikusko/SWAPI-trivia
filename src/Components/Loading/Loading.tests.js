import React from 'react';
import { shallow } from 'enzyme';
import Loading from './Loading';
