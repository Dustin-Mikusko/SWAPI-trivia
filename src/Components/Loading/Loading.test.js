import React from 'react';
import { shallow } from 'enzyme';
import Loading from './Loading';

describe('Loading', () => {
  const wrapper = shallow(<Loading />);

  it('should render loading component to match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
