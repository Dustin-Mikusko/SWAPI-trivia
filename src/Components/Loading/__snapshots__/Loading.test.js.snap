// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render loading component to match snapshot 1`] = `
<main
  className="loader"
>
  <div
    className="title-container"
  >
    <p
      className="loading-title loading"
    >
      Gathering data from a galaxy far, 
      <br />
      far away
    </p>
  </div>
</main>
`;
